[Edit Tutor’s profile] Verify that the Tutor can not change the password with invalid data #2770

lizabre opened the pull request:
Click on the “Change password” button.
Expected Results:
The mock-up “Change password” is opened.

Fill in the field “Current password” by current password.

Fill in the following fields for invalid data: “New Password”, Re-type new password”.

“A1!”
“ABCDEFGHIJKabcdefghijk1234567890!@#$%^&*()_+?><”
“ABab12!”
“ABCDEFGabcdefg123456!@#$%^”
Click on the button “Save the password”
Expected Results:
The error messages will appear.

**Test result:**
![image](https://github.com/user-attachments/assets/71eb4e7a-fe12-49f8-a97f-d529b28b1d31)

lizabre committed:
added verififcation for invalid data in ChangePasswordModal

lizabre committed:
add correct test

lizabre committed:
small fixes

luvthenika committed:
fixed issue with props types in DividerDropdonwAutoComplete (#3335)

* fixed issue with props types in asyncAutoComplete

* fixed linter error

* fixed linter error

* fix: format code with Prettier

* fixed vite version

* fixed vite version

irynalaitaruk committed:
Fixed styles in AddCourseTemplate modal #3312 (#3345)

* Fixed styles in AddCourseTemplate modal #3312

Iarynovskyi commented:
Just curious
Why do you have 3 branches in _bugfix/2770/.._ for this task ? 
Or am I misunderstanding something ?

If it is not on purpose, you will have to delete them yourself after the merge
Because only the one from which the PR was dropped will be automatically deleted

lizabre commented:
> Just curious Why do you have 3 branches in _bugfix/2770/.._ for this task ? Or am I misunderstanding something ?
> 
> If it is not on purpose, you will have to delete them yourself after the merge Because only the one from which the PR was dropped will be automatically deleted

Had some troubles with making branches, took a few tries to settle everything up
Thank you for reminding me, I will delete the ones that are not in use!

IrynaKhylchuk committed:
Implemented tests for "Last name" (#3353)

IrynaKhylchuk committed:
[Edit profile. Profile] Verify that the tutor/student couldn't save the first name only with space characters (#3336)

* Added test for EditProfile component

* added tests

* invalid test removed

VKormylo committed:
Fixed linter issue (#3389)

luvthenika committed:
Fixed translation in Computer Science Subject (#3380)

* added field into tr

* removed env

* fix linter

* fix ukranian spelling

IrynaKhylchuk committed:
Fixed incorrect URL (#3390)

VKormylo committed:
Replaced useAxios with useQuery and useMutation in UserTable component (#3363)

* Updated endpoints and changed interface to a type

* Updated some functions in user-service

* Replaced useAxios with useQuery and useMutation in UserTable component

* Updated unit tests for user-service and UserTable component

* Fixed sonar issue with unit test

* Added onSuccess and onError to useMutation, updated test case

* Fixed comments

Iarynovskyi committed:
Replace useAxios with useMutation in CreateSubjectModal  (#3356)

* add useMutation and createSubject in subject-service for CreateSubjectModal

* add some tests for changed files

* rename in import

* fix comment from SonarBot

* cleaning

* fix comments

* changes in tests

* hid from me but I found it

* another changes

* return vi

yur4uwe committed:
Test that ensures that user can save last name up to 30 characters included   (#3370)

* test created, also fixed little grammatical errors

* tests are just an illusion created to cater our peace

lizabre committed:
added verification for error at just alpabetic values (#3350)

added verification for error at just alpabetic values in fields "current password" and "new password"

VKormylo committed:
Renamed user-service.tsx to .ts (#3395)

oleg191006 committed:
(SP:) [Edit profile.Profile] Verify that the tutor/student can save a First Name containing alphabetic characters from Cyrillic and Latin scripts (#3361)

oleg191006 committed:
add validation test for 'Last name' input with invalid characters (#3355)

oleg191006 committed:
(SP:) [Edit profile. Professional info] Verify that the user can delete a subject in the "Your professional category" pop-up.  (#3352)

yur4uwe committed:
Added tests for First name input (#3358)

* implemented test for error message to appear when user input unfitting characters for First or last name

* added test to ensure that all listed valid chracters are accepted and deleted check for last name as its done by another contributor

oleg191006 committed:
replace getByLabelText to getByPlaceholderText (#3404)

IrynaKhylchuk committed:
Fixed error when getting by resourse id (#3393)

VKormylo committed:
Added react-query to userProfileLoader (#3347)

* Updated loader function so that it prefetches the query for user

* Updated handle function for user-profile route to use react-query

* Added unit tests for userProfileLoader and userProfile handle

lizabre committed:
(SP: ) Remove starting modal when there is no time limit on quiz #3366 (#3381)

* removed TimeLimitReminder for games without time limit

* fixed types

* fixed styles

lizabre committed:
resolved conflict
